__device-tests__/helpers/utils.js

```diff
@@ -118,10 +126,92 @@ const stopDriver = async ( driver: wd.PromiseChainWebdriver ) => {
 	}
 };
 
+// attempts to type a string to a given element, need for this stems from
+// https://github.com/appium/appium/issues/12285#issuecomment-471872239
+// https://github.com/facebook/WebDriverAgent/issues/1084
+const typeString = async ( driver: wd.PromiseChainWebdriver, element: wd.PromiseChainWebdriver.Element, str: string, clear: boolean = false ) => {
+	if ( clear ) {
+		await element.clear();
+	}
+
+	if ( isAndroid() ) {
+		if ( str in strToKeycode ) {
+			return await driver.pressKeycode( strToKeycode[ str ] );
+		}
+		const paragraphs = str.split( '\n' );
+
+		if ( paragraphs.length > 1 ) {
+			for ( let i = 0; i < paragraphs.length; i++ ) {
+				const paragraph = paragraphs[ i ].replace( /[ ]/g, '%s' );
+				if ( paragraph in strToKeycode ) {
+					await driver.driver.pressKeycode( strToKeycode[ str ] );
+				} else {
+					await driver.execute( 'mobile: shell', { command: 'input', args: [ 'text', paragraph ] } );
+				}
+				if ( i !== paragraphs.length - 1 ) {
+					await driver.pressKeycode( strToKeycode[ '\n' ] );
+				}
+			}
+		} else {
+			await driver.execute( 'mobile: shell', { command: 'input', args: [ 'text', str.replace( /[ ]/g, '%s' ) ] } );
+		}
+	} else {
+		return await element.type( str );
+	}
+};
+
+// Calculates middle x,y and clicks that position
+const clickMiddleOfElement = async ( driver: wd.PromiseChainWebdriver, element: wd.PromiseChainWebdriver.Element ) => {
+	const location = await element.getLocation();
+	const size = await element.getSize();
+
+	const action = await new wd.TouchAction( driver );
+	action.press( { x: location.x + ( size.width / 2 ), y: location.y } );
+	action.release();
+	await action.perform();
+};
+
+// Clicks in the top left of an element
+const clickBeginningOfElement = async ( driver: wd.PromiseChainWebdriver, element: wd.PromiseChainWebdriver.Element ) => {
+	const location = await element.getLocation();
+	const action = await new wd.TouchAction( driver );
+	action.press( { x: location.x, y: location.y } );
+	action.release();
+	await action.perform();
+};
+
+// Starts from the middle of the screen or the element(if specified)
+// and swipes upwards
+const swipeUp = async ( driver: wd.PromiseChainWebdriver, element: wd.PromiseChainWebdriver.Element = undefined ) => {
+	let size = await driver.getWindowSize();
+	let y = 0;
+	if ( element !== undefined ) {
+		size = await element.getSize();
```

Comment:
I feel this function could use an upgrade (maybe in another PR?) It seems to me that there are quite few edge cases where this approach would fail:
- What if the `element` height is bigger than 3 times the height of the screen
- What if `element` is not given and a single swipe is not enough
- What if the screen is only showing the top of `element`? `startY ` might be outside the screen boudaries. Also a swipe might not be enough to reveal the bottom of the element in this case

I think a "slightly" better approach would be to make this function a simple swipe up with an height parameter, it will try to center the swipe on the screen and do it in a loop if height is bigger than the size of the screen. We can also think of a `swipeUpUntilElementIsVIsible()` version which would do what its name says.

Reply:
For sure! I was saving improving this for another PR, do you think it's good enough for now? 

>  it will try to center the swipe on the screen

It originally just did this, but failed on the CI because the device it used was small and the keyboard took up more than half of the screen. 

> We can also think of a swipeUpUntilElementIsVIsible() version which would do what its name says.

Yep, for now I was hoping this was enough to use in a loop if needed before that get's implemented.

Reply:
Like what's done hear, https://github.com/wordpress-mobile/gutenberg-mobile/pull/821/files#diff-5c1289e37617de9b4adbba51cf493004R136

Reply:
Yep it seems to work really smoothly for now (at least on the devices we tested :)). So I think we should be good 👍